### ISSUE TITLE
Missing libjvmtiasmagent.so when using profrecord

### ISSUE THREAD

**Issue by itarato** [graalvm/mx#274]
Following the https://github.com/graalvm/mx/blob/master/README-proftool.md doc page I was trying to run `profrecord`:

```bash
./mx profrecord -E ~/Desktop/profrecord -O jt ruby -e "10.times.to_a"
```

Fails with:

```bash
./mx/mxbuild/linux-amd64/com.oracle.jvmtiasmagent/amd64/libjvmtiasmagent.so hasn't been built yet
```

The file however exists in:

```bash
./mx/mxbuild/linux-amd64-jdk19/com.oracle.jvmtiasmagent/amd64/libjvmtiasmagent.so
```

If I copy that file to the folder it's missing from, the `profrecord` operation works fine.

`mx` HEAD on my local:

```
commit 0280b223a74747e31cb4a7d603263d6d69c65fdb (HEAD -> master, tag: 6.20.3, origin/master, origin/HEAD)
Date:   Tue Apr 25 15:44:33 2023 +0000
```

Doing `mx build` is not helping.

**Comment by tkrodriguez** [comment]
The problem appears to be a poor error message when `mx profrecord` is run without a configured `JAVA_HOME`.  This will really only happen when `profrecord` is used in a standalone way.  I'll improve the error reporting to make it more clear what has gone wrong.

**Comment by tkrodriguez** [comment]
This should be fixed by fb4f77b0cff81bcee2e93b30a40cb3e6fbb94848.  I was unable to link the PR for some reason.
